### PR TITLE
Storage: Use WithSnapshot for Files

### LINKED ISSUE #8076: [P0] Storage: Add `WithSnapshot` to Files client types
https://apiview.azurewebsites.net/Assemblies/Review/47c9ebbff8c0458fa2e7904eae34656f

And remove the various snapshot parmaeters

### PR DESCRIPTION
**Pull request by tg-msft**
Fixes #8076

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
Have you re-rerun the related live tests?

File: sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs [hunk 1]
```diff
@@ -676,15 +678,13 @@ private void SetNameFieldsIfNull()
...
-                    $"{nameof(Uri)}: {Uri}\n" +
-                    $"{nameof(shareSnapshot)}: {shareSnapshot}");
+                    $"{nameof(Uri)}: {Uri}");
                 try
                 {
                     Response<RawStorageDirectoryProperties> response = await FileRestClient.Directory.GetPropertiesAsync(
                         ClientDiagnostics,
                         Pipeline,
                         Uri,
-                        sharesnapshot: shareSnapshot,
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs]: Shouldn't this line remain, and be changed to ```sharesnapshot: this.Snapshot```?

File: sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs [hunk 2]
```diff
@@ -1109,7 +1096,6 @@ private void SetNameFieldsIfNull()
                         marker: marker,
                         prefix: prefix,
                         maxresults: pageSizeHint,
-                        sharesnapshot: shareSnapshot,
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs]: same as above

File: sdk/storage/Azure.Storage.Files/src/FileClient.cs [hunk 3]
```diff
@@ -1218,15 +1204,13 @@ private void SetNameFieldsIfNull()
...
-                    $"{nameof(Uri)}: {Uri}\n" +
-                    $"{nameof(shareSnapshot)}: {shareSnapshot}");
+                    $"{nameof(Uri)}: {Uri}");
                 try
                 {
                     Response<RawStorageFileProperties> response = await FileRestClient.File.GetPropertiesAsync(
                         ClientDiagnostics,
                         Pipeline,
                         Uri,
-                        sharesnapshot: shareSnapshot,
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files/src/FileClient.cs]: same as above

File: sdk/storage/Azure.Storage.Files/src/FileClient.cs [hunk 4]
```diff
@@ -2235,15 +2205,13 @@ private void SetNameFieldsIfNull()
...
-                    $"{nameof(Uri)}: {Uri}\n" +
-                    $"{nameof(shareSnapshot)}: {shareSnapshot}");
+                    $"{nameof(Uri)}: {Uri}");
                 try
                 {
                     return await FileRestClient.File.GetRangeListAsync(
                         ClientDiagnostics,
                         Pipeline,
                         Uri,
-                        sharesnapshot: shareSnapshot,
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files/src/FileClient.cs]: same as above

**Review by seanmcc-msft — CHANGES_REQUESTED** [review]
Do we not have any unit tests for file or directory snapshots?

File: sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs (hunk 1 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs]: No, the snapshot is passed along in the `Uri`.  There is no `this.Snapshot` to pass.

File: sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs (hunk 2 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs]: Ditto

File: sdk/storage/Azure.Storage.Files/src/FileClient.cs (hunk 3 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Files/src/FileClient.cs]: Ditto

File: sdk/storage/Azure.Storage.Files/src/FileClient.cs (hunk 4 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Files/src/FileClient.cs]: Ditto

**Comment by tg-msft** [comment]
No - there isn't meaningful test coverage.  We create a snapshot and verify that calling WithSnapshot updates the Uri.

**Review by seanmcc-msft — APPROVED** [review]

**Review by amnguye — APPROVED** [review]

**Comment by tg-msft** [comment]
Yes, I've run live tests for Files.

**Pull request merged**